src/vos/vos_aggregate.c

```diff
@@ -1244,7 +1244,7 @@ fill_one_segment(daos_handle_t ih, struct agg_merge_window *mw,
 }
 
 static int
-fill_segments(daos_handle_t ih, struct agg_merge_window *mw,
+fill_segments(daos_handle_t ih, struct agg_merge_window *mw, struct umem_instance *umm,
```

Comment:
Alternatively, you could pass agg_param which has a pointer to the merge window and the umm

Reply:
ok, changed it together with rebase (the last commit), thx